edc-extensions/business-partner-validation2/src/main/java/org/eclipse/tractusx/edc/validation/businesspartner/BusinessPartnerEvaluationExtension.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.tractusx.edc.validation.businesspartner;
+
+import org.eclipse.edc.policy.engine.spi.PolicyEngine;
+import org.eclipse.edc.policy.engine.spi.RuleBindingRegistry;
+import org.eclipse.edc.policy.model.Permission;
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.tractusx.edc.validation.businesspartner.functions.BusinessPartnerGroupFunction;
+import org.eclipse.tractusx.edc.validation.businesspartner.spi.BusinessPartnerGroupStore;
+
+import static org.eclipse.edc.connector.contract.spi.offer.ContractDefinitionResolver.CATALOGING_SCOPE;
+import static org.eclipse.edc.connector.contract.spi.validation.ContractValidationService.NEGOTIATION_SCOPE;
+import static org.eclipse.edc.connector.contract.spi.validation.ContractValidationService.TRANSFER_SCOPE;
+
+/**
+ * Registers a {@link org.eclipse.tractusx.edc.validation.businesspartner.functions.BusinessPartnerGroupFunction} for the following scopes:
+ * <ul>
+ *     <li>{@link org.eclipse.edc.connector.contract.spi.offer.ContractDefinitionResolver#CATALOGING_SCOPE}</li>
+ *     <li>{@link org.eclipse.edc.connector.contract.spi.validation.ContractValidationService#NEGOTIATION_SCOPE}</li>
+ *     <li>{@link org.eclipse.edc.connector.contract.spi.validation.ContractValidationService#TRANSFER_SCOPE}</li>
+ * </ul>
+ * The rule to which the function is bound is {@link BusinessPartnerEvaluationExtension#BUSINESS_PARTNER_CONSTRAINT_KEY}. That means, that policies that are bound to these scopes look
+ * like this:
+ * <pre>
+ * {
+ *     "constraint": {
+ *         "leftOperand": "BusinessPartnerGroup",
```

Comment:
We should be careful here and namespace the operand value because ODRL defines leftOperand as an IRI. This will result in expansion either placing it in the vocabulary specified by a prefix or, if one is not specified, in the default EDC namespace.

Reply:
wouldn't the default namespace be the TX one (`tx:https://w3id.org/tractusx/v0.0.1/ns/`)?

Reply:
not sure if it is C-X or T-X, I always get confused. @wolf4ood

Reply:
currently, we have this: 
```java
    public static final String TX_PREFIX = "tx";
    public static final String TX_NAMESPACE = "https://w3id.org/tractusx/v0.0.1/ns/";
```